enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
-            cls (type): The EnterpriseCustomerManageLearnersView class itself
-            enterprise_customer: The instance of EnterpriseCustomer whose attached users we're enrolling
-            emails: An iterable of strings containing email addresses to enroll in a course
-            course_id: The ID of the course in which we want to enroll
-            mode: The enrollment mode the users will be enrolled in the course with
-            request: The HTTP request the enrollment is being created by
-            notify: Whether to notify (by email) the users that have been enrolled
+            enterprise_customer: The EnterpriseCustomer which is sponsoring the enrollment
+            user: The user who needs to be enrolled in the course
+            course_mode: The mode with which the enrollment should be created
+            *course_ids: An iterable containing any number of course IDs to eventually enroll the user in.
+
+        Returns:
+            Boolean: Whether or not enrollment succeeded for all courses specified
         """
-        enrolled = []
-        non_existing = []
-        failed = []
+        enterprise_customer_user, __ = EnterpriseCustomerUser.objects.get_or_create(
+            enterprise_customer=enterprise_customer,
+            user_id=user.id
+        )
         enrollment_client = EnrollmentApiClient()
-        course_details = CourseCatalogApiClient(request.user).get_course_run(course_id)
+        succeeded = True
+        for course_id in course_ids:
+            try:
+                enrollment_client.enroll_user_in_course(user.username, course_id, course_mode)
+            except HttpClientError as exc:
+                succeeded = False
+                error_message = json.loads(exc.content.decode()).get('message', 'No error message provided')
+                logging.error(
+                    'Error while enrolling user %(user)s: %(message)s',
+                    dict(user=user.username, message=error_message)
+                )
+            else:
+                EnterpriseCourseEnrollment.objects.get_or_create(
+                    enterprise_customer_user=enterprise_customer_user,
+                    course_id=course_id
+                )
+        return succeeded
+
+    @classmethod
+    def get_users_by_email(cls, emails):
+        """
+        Accept a list of emails, and separate them into users that exist on OpenEdX and users who don't.
+
+        Args:
+            emails: An iterable of email addresses to split between existing and nonexisting
+
+        Returns:
+            users: Queryset of users who exist in the OpenEdX platform and who were in the list of email addresses
+            missing_emails: List of unique emails which were in the original list, but do not yet exist as users
+        """
+        users = User.objects.filter(email__in=emails)
+        present_emails = users.values_list('email', flat=True)
+        missing_emails = list(set(emails) - set(present_emails))
```

Comment:
I know, but a set isn't analogous to a queryset, which is the other return value. I'd rather convert to list to avoid any inadvertent future goofs due to unexpected behavior.